ui/js/components/elements/tree.jsx

```diff
@@ -38,34 +38,40 @@ d3.selection.prototype.tspans2 = function(lines, lh) {
 }
 
 
-d3.wordwrap = function (line, maxCharactersPerLine, maxLines = 2) {
-	var parts = line.split(/([^a-z][a-z]*)/), //split at non-words
+d3.wordwrap = function (line = '', maxCharactersPerLine = 43, maxLines = 3) {
+	const regexChecks = [/_/ig, /-/ig, /([A-Z])/g];
+	var parts = [],
 		lines = [],
-		words = [],
-		maxChars = maxCharactersPerLine || 40
+		words = [];
+
+	for(let num in regexChecks) {
+		if (line.search(regexChecks[num]) >= 0) {
+			parts = line.replace(regexChecks[num], (num == 2 ? ' $1' : ' ')).split(' ');
```

Comment:
This will overwrite the array that parts was instantiated as, and will not accumulate more than a single regex replacement - This should instead be something like `parts.push(...` - Also might be a little cleaner to wrap this loop up as a reduce rather than a for loop.